src/main.js

```diff
@@ -1,6 +1,58 @@
-import { example } from './data.js';
-// import data from './data/lol/lol.js';
+import { filterData, sortData, getArrayUnique, computeStats } from './data.js';
 import data from './data/pokemon/pokemon.js';
-// import data from './data/rickandmorty/rickandmorty.js';
+const root = document.getElementById('root')
+const inputUser = document.querySelector('#input-user');
+const selectOrder = document.querySelector('#select-order');
+const selectType = document.querySelector('#select-type');
+const estadistics = document.querySelector('#estadistics');
 
-console.log(example, data);
+
+inputUser.addEventListener('keyup', (e) => {
+  const arrayFiltered = filterData(data.pokemon, 'name', e.target.value);
+  renderView(arrayFiltered);
+})
+
+selectOrder.addEventListener('change', (e) => {
+  const arrayOrdered = sortData(data.pokemon, 'name', e.target.value);
+  renderView(arrayOrdered);
+})
+
+selectType.addEventListener('change', (e) => {
+  const arrayType = filterData(data.pokemon, 'type', e.target.value);
+  renderView(arrayType);
+})
+
+estadistics.addEventListener('click', () => {
+  root.innerHTML = `
+                    <p>Este es el promedio de base attack ${computeStats(data.pokemon, 'base-attack')}</p>
+                    <p>Este es el promedio de base defense ${computeStats(data.pokemon, 'base-defense')}</p>
+                    <p>Este es el promedio de base stamina ${computeStats(data.pokemon, 'base-stamina')}</p>
+                    <p>Este es el promedio de max-cp ${computeStats(data.pokemon, 'max-cp')}</p>
+                    <p>Este es el promedio de max-cp ${computeStats(data.pokemon, 'max-hp')}</p>
+                    `;
+})
+
+
+const renderView = (arrayData) => {
+  let html = `<div><ul class='wrap'>`;
+  arrayData.map(element => {
+    html += `<li class= "card">
+              <img src=${element.img}>
+              <p>${element.name}</p>
+             </li>
+            `
+  });
+  html += `<ul></div>`
+  root.innerHTML = html;
```

Comment:
no concatenamos `html` dentro de map. usamos el arreglo que devuelva map

const 
```suggestion
  const itemsHTML = arrayData.map(el => `<li class= "card">
        <img src=${el.img}>
        <p>${el.name}</p>
   </li>`);
  root.innerHTML = `<ul class='wrap'>${itemsHTML.join('')}<ul>`;
```